### PR TITLE
chore(src/icons): add Chevron Icons

### PR DESCRIPTION
**Pull request by shridhar52war**

### CODE REVIEW

File: src/icons/ChevronDown/ChevronDown.native.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        fill={fill}
+      />
+    </Svg>
+  );
+}
+
+ChevronDown.propTypes = {
+  width: PropTypes.number,
+  height: PropTypes.number,
+  fill: PropTypes.oneOf(getThemeColors()),
```
Comment (deorukhkarchaitanya) [inline src/icons/ChevronDown/ChevronDown.native.js]: Should we accept theme color keys here instead? Like primary.800 & then resolve the color on the component? Not sure about this. Lmk what you think.

**Comment by deorukhkarchaitanya** [comment]
LGTM

File: src/_helpers/theme.js [hunk 2]
```diff
@@ -46,4 +46,16 @@ const getLineHeight = (currentTheme, textSize) => {
   }
 };
 
-export { makePxValue, getColorKeys, getColor, getVariantColorKeys, getLineHeight };
+const getThemeColors = () => {
+  const flatten = (obj) => {
```
Comment (dhruvdutt) [inline src/_helpers/theme.js]: Flatten can't be implemented inside `theme.js`. Put inside `_helpers/flatten.js`.

File: src/icons/ChevronDown/ChevronDown.native.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+};
+
+ChevronDown.defaultProps = {
+  width: 24,
+  height: 24,
+  fill: 'rgba(43, 131, 234, 1.0 )',
```
Comment (dhruvdutt) [inline src/icons/ChevronDown/ChevronDown.native.js]:
```suggestion
  fill: 'rgba(43, 131, 234, 1.0)',
```

File: src/_helpers/theme.js (hunk 2 above)
Comment (shridhar52war) [reply src/_helpers/theme.js]: This is not exactly flattening, may be I named the function wrong. It just gets the deepest nested value. Not sure what name should it be given.

File: src/icons/ChevronDown/ChevronDown.web.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+ChevronDown.propTypes = {
+  width: PropTypes.number,
+  height: PropTypes.number,
+  fill: PropTypes.oneOf(getThemeColors()),
+};
+
+ChevronDown.defaultProps = {
+  width: 24,
+  height: 24,
+  fill: 'rgba(43, 131, 234, 1.0 )',
```
Comment (dhruvdutt) [inline src/icons/ChevronDown/ChevronDown.web.js]:
```suggestion
  fill: 'rgba(43, 131, 234, 1.0)',
```

File: src/icons/ChevronDown/ChevronDown.native.js (hunk 3 above)
Comment (dhruvdutt) [reply src/icons/ChevronDown/ChevronDown.native.js]: Also, color codes shouldn't be hardcoded like this in components.

File: src/icons/ChevronLeft/ChevronLeft.native.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+ChevronLeft.propTypes = {
+  width: PropTypes.number,
+  height: PropTypes.number,
+  fill: PropTypes.oneOf(getThemeColors()),
+};
+
+ChevronLeft.defaultProps = {
+  width: 24,
+  height: 24,
+  fill: 'rgba(43, 131, 234, 1.0 )',
```
Comment (dhruvdutt) [inline src/icons/ChevronLeft/ChevronLeft.native.js]:
```suggestion
  fill: 'rgba(43, 131, 234, 1.0)',
```

File: src/icons/ChevronLeft/ChevronLeft.web.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+ChevronLeft.propTypes = {
+  width: PropTypes.number,
+  height: PropTypes.number,
+  fill: PropTypes.oneOf(getThemeColors()),
+};
+
+ChevronLeft.defaultProps = {
+  width: 24,
+  height: 24,
+  fill: 'rgba(43, 131, 234, 1.0 )',
```
Comment (dhruvdutt) [inline src/icons/ChevronLeft/ChevronLeft.web.js]:
```suggestion
  fill: 'rgba(43, 131, 234, 1.0)',
```

File: src/icons/ChevronRight/ChevronRight.native.js [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+ChevronRight.propTypes = {
+  width: PropTypes.number,
+  height: PropTypes.number,
+  fill: PropTypes.oneOf(getThemeColors()),
+};
+
+ChevronRight.defaultProps = {
+  width: 24,
+  height: 24,
+  fill: 'rgba(43, 131, 234, 1.0 )',
```
Comment (dhruvdutt) [inline src/icons/ChevronRight/ChevronRight.native.js]:
```suggestion
  fill: 'rgba(43, 131, 234, 1.0)',
```

File: src/icons/ChevronRight/ChevronRight.web.js [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+ChevronRight.propTypes = {
+  width: PropTypes.number,
+  height: PropTypes.number,
+  fill: PropTypes.oneOf(getThemeColors()),
+};
+
+ChevronRight.defaultProps = {
+  width: 24,
+  height: 24,
+  fill: 'rgba(43, 131, 234, 1.0 )',
```
Comment (dhruvdutt) [inline src/icons/ChevronRight/ChevronRight.web.js]:
```suggestion
  fill: 'rgba(43, 131, 234, 1.0)',
```

File: src/icons/ChevronUp/ChevronUp.native.js [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+ChevronUp.propTypes = {
+  width: PropTypes.number,
+  height: PropTypes.number,
+  fill: PropTypes.oneOf(getThemeColors()),
+};
+
+ChevronUp.defaultProps = {
+  width: 24,
+  height: 24,
+  fill: 'rgba(43, 131, 234, 1.0 )',
```
Comment (dhruvdutt) [inline src/icons/ChevronUp/ChevronUp.native.js]:
```suggestion
  fill: 'rgba(43, 131, 234, 1.0)',
```

File: src/icons/ChevronUp/ChevronUp.web.js [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+ChevronUp.propTypes = {
+  width: PropTypes.number,
+  height: PropTypes.number,
+  fill: PropTypes.oneOf(getThemeColors()),
+};
+
+ChevronUp.defaultProps = {
+  width: 24,
+  height: 24,
+  fill: 'rgba(43, 131, 234, 1.0 )',
```
Comment (dhruvdutt) [inline src/icons/ChevronUp/ChevronUp.web.js]:
```suggestion
  fill: 'rgba(43, 131, 234, 1.0)',
```

File: src/icons/ChevronDown/ChevronDown.native.js (hunk 3 above)
Comment (shridhar52war) [reply src/icons/ChevronDown/ChevronDown.native.js]: How about `fill: colors.sapphire[800]`, where colors is imported from `"/tokens/colors"`

Comment (dhruvdutt) [reply src/icons/ChevronDown/ChevronDown.native.js]: Yes, that would make more sense.

File: src/_helpers/theme.js (hunk 2 above)
Comment (dhruvdutt) [reply src/_helpers/theme.js]: Is this needed after changing prop to `fill: colors.sapphire[800]`?

**Comment by dhruvdutt** [comment]
Are we not adding stories for icons?

File: src/_helpers/theme.js (hunk 2 above)
Comment (shridhar52war) [reply src/_helpers/theme.js]: Yes, we will need this. `fill: colors.sapphire[800]` is the default value, where as fill can be any value from `getThemeColors` i.e `fill: PropTypes.oneOf(getThemeColors())`.

**Comment by shridhar52war** [comment]
> Are we not adding stories for icons?
There is already a story for `Icon` component.

File: src/icons/ChevronDown/ChevronDown.native.js (hunk 1 above)
Comment (kamleshchandnani) [reply src/icons/ChevronDown/ChevronDown.native.js]: Yes why can't we do this?

File: src/_helpers/theme.js [hunk 11]
```diff
@@ -46,4 +46,16 @@ const getLineHeight = (currentTheme, textSize) => {
   }
 };
 
-export { makePxValue, getColorKeys, getColor, getVariantColorKeys, getLineHeight };
+const getThemeColors = () => {
```
Comment (kamleshchandnani) [inline src/_helpers/theme.js]: why do we need this? we can just say `azzure.800` or anything like that.

File: src/icons/ChevronDown/ChevronDown.native.js [hunk 12]
```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Svg, { Path } from 'react-native-svg';
+import { getThemeColors } from '../../_helpers/theme';
+import colors from '../../tokens/colors';
+
+export default function ChevronDown({ width, height, fill }) {
```
Comment (kamleshchandnani) [inline src/icons/ChevronDown/ChevronDown.native.js]:
```
const ChevronDown = ({ width, height, fill }) => {}

export default ChevronDown
```

File: src/icons/ChevronDown/ChevronDown.native.js (hunk 1 above)
Comment (shridhar52war) [reply src/icons/ChevronDown/ChevronDown.native.js]: `fill` comes from `Icon` component which is resolved from the theme it self. All the components in the `/icons` folder (namely `checkboxFilled`, `checkboxOutlined`) accept the derived values. So accepting a theme color like "primary.900" here, then again making use of theme context to derive the value doesn't make sense to me.
@deorukhkarchaitanya

Comment (deorukhkarchaitanya) [reply src/icons/ChevronDown/ChevronDown.native.js]: Yeah. These components are in a way ```assests```. Passing theme here would be odd.

Comment (deorukhkarchaitanya) [reply src/icons/ChevronDown/ChevronDown.native.js]: It's like asking designers to give us Svgs with theme aliases

**Pull request merged**